aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -46,9 +51,48 @@ module AptosFramework::Token {
         name: ASCII::String,
     }
 
+
+    /// The royalty of a token
+    struct Royalty has copy, drop, store {
+        // The basis point and denominator is ROYAL_BASIS_POINT_DENOMINATOR (1,000,000)
+        royalty_points_per_million: u64,
```

Comment:
decimal in coin is for getting a user representation of coin amount. Here, we need both nominator and denominator so that we can represent a percentage. If you want the denominator to be flexible. we can make it a variable entered by the users. However, it is better to keep the denominator a constant across different tokens. So the royal points (nominator) are comparable across different tokens, eg:  Metaplex assuming the denominator is 100 in code